src/modules/75-cf/pages/feature-flags/components/FeatureFlagsListing.tsx

```diff
@@ -63,18 +64,38 @@ const FeatureFlagsListing: FC<FeatureFlagsListingProps> = ({
 }) => {
   const gitSync = useFFGitSyncContext()
   const { getString } = useStrings()
-
+  const [checkedFlags, setCheckedFlags] = useState<Feature[]>([])
+  const [selectAll, setSelectAll] = useState(false)
   const [flagFilter] = useQueryParamsState<Optional<FilterProps>>('filter', {})
 
-  const onCheckboxSelect = useCallback((event: FormEvent<HTMLInputElement>) => {
-    if (event.currentTarget.checked) {
-      //TODO: Handle checked
+  const { FFM_8344_FLAG_CLEANUP } = useFeatureFlags()
+
+  const isStaleFlagsView =
+    FFM_8344_FLAG_CLEANUP && flagFilter?.queryProps?.value === FeatureFlagStatus.POTENTIALLY_STALE
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const handleCheckboxChange = (e: any, rowData: any): void => {
+    const { checked } = e.target
+    if (checked) {
+      const tempChecked = [...checkedFlags]
+      tempChecked.push(rowData.value.identifier)
+      setCheckedFlags(tempChecked)
```

Comment:
You could shorten this and do away with the `tempChecked` const if no other operations are needed here
```suggestion
      setCheckedFlags([...checkedFlags, rowData.value.identifier)
```